### ISSUE TITLE
Face Verification Notebook

### ISSUE THREAD

**Issue by maralski** [Bren077s/SideProjects#1]
I read your notebook on face verification. I had a few thoughts.

- Since there are insufficient samples to generalise I wonder whether adding more regularisation to the network will help. This should reduce the number of parameters in the network, i.e. adjust dropout.
- Reset a portion of pre-trained weights to zero using some threshold to achieve a reduction in pre-trained parameters.
- For each image in training set take the weights from one of the last few layers in the network and use it as a set of embeddings then compare distance between embeddings to sort and rank images for classification.

**Comment by bhansconnect** [comment]
For your first point, here is definitely a decent amount of improvements that could go into this face verification. If you look "What Next?" section, I mentioned some of your improvements among others. I think the main way to make it better would simply be to load more samples. That being said, This was only a proof of concept so I am not planning on putting a ton of processing power into it.

As for your second point, I have never thought of doing something like that. It sounds like a very interesting idea. I will have to try it at some point. Is there any research paper about the results of doing that?

And for your third point, if I really wanted to make it state of the art, I would model after [Facenet](https://arxiv.org/abs/1503.03832v3). It has an advanced embedding system especially for training.

**Comment by maralski** [comment]
I had a brief read of Facenet and its the same idea as using an autoencoder method to produce embeddings. As for the second point I have not seen any papers that explore this idea. It would require some experiments.

**Comment by bhansconnect** [comment]
It is a similar idea to autoencoders, but the triplet loss function is what enables it to learn so well.
